content/general/limits.textile

```diff
@@ -0,0 +1,306 @@
+
+|_=. Limit |_=. Free |_=. Self-service |_=. Business |_=. Enterprise |
+| "Number of subscribers":#subscribers (per channel) |\4=. Unlimited |
+| "Presence members":#presence-members (per channel) | 50 | 200 | 200 | Custom |
+| "Channel creation rate":#channel-create (per second) | 20 | >=50 | >=50 | Custom |
+| "Message publish rate":#message-publish-rate (per channel) | 15 | 50 | 50 | Custom |
+
+h3(#subscribers). Number of subscribers
+
+There is no limit on the number of clients that can be subscribed to a channel.
+
+h3(#presence-members). Presence members
+
+The number of clients that can use the "presence feature":/realtime/presence in a channel is limited as a precaution. If a large number of clients are subscribed to presence events on a channel then the amount of messages can scale very quickly.
+
+As an example, consider 200 clients subscribed to presence events on a channel and all of them join and leave the presence set within a few minutes. This would result in the following messages:
+
+* 200 presence messages published for the enter event.
+* 200 x 200 (40,000) messages subscribed to for the enter events.
+* 200 presence messages published for the leave event.
+* 200 x 200 (40,000) presence messages subscribed to for the leave event.
+
+This highlights the potential for 80,400 messages to be sent in a very short space of time on a single channel.
+
+Any clients that attempt to join the presence set over the presence member limit will be rejected and the error code @91003@ will be returned.
+
+h3(#channel-create). Channel creation rate
+
+The channel creation rate is the number of channels that can be created per second.
+
+The limit is calculated based on the hard limit for "peak channels":#peak-channels, with a minimum value of 50 per second for self-service and business accounts.
+
+If the channel creation rate is exceeded, an "instantaneous rate limit":#instantaneous-rates will be applied to channel creation.
+
+h3(#message-publish-rate). Message publish rate
+
+The message rate is the maximum rate at which messages can be published on a channel.
+
+If the message publish rate is exceeded for a channel, an "instantaneous rate limit":#instantaneous-rates on publishing will be applied to that channel.
+
+h2(#message). Message limits
+
+Message limits relate to the number, rate and bandwidth of "messages":/realtime/messages on your account.
+
+|_=. Limit |_=. Free |_=. Self-service |_=. Business |_=. Enterprise |
+| "Message size":#message-size (KiB) | 16 | 64 | 64 | 256 |
+| "History TTL":#history-ttl (hours) | 24 | 72 | 72 | Custom |
+
+h3(#message-size). Message size
+
+The message size is the maximum size of a single published "message":/glossary#messages.
+
+Note that the average message size is set to 2KiB. Although a message can be published that is 16KiB, this will count as 8 messages towards your "message count":#messages.
```

Comment:
Maybe talk about a "maximum message size of 16 KiB" here, for clarity?

Reply:
I've re-worded this as I think it was confusing. My understanding is that every 2KiB of message counts as a 1 message towards the message count. Technically a single message can be sent at 16KiB, but it would "count" as 8 messages. Hopefully the re-word makes this clearer.

Reply:
That's not quite correct. Messages are counted, and the payload sizes are aggregated, over a billing period. Then, the mean message size is calculated for the period (which is the aggregate message payload size divided by the count of messages). If this is 2KiB or below, there are no additional message charges. So, messages whose payload is larger than 2KiB only trigger additional messages being counted if the set of messages as a whole have a mean payload larger than 2KiB. If there is a 16KiB message, followed by 15 1KiB messages, then the mean will be below 2KiB and there will be no additional charge.

Reply:
And ... I believe 16KiB is only the maximum for free accounts. Other accounts can have higher limits.

Reply:
Ah, thanks for the clarification on that Paddy. 

I made some notes on "what counts as a message" in a separate issue whilst reading up on limits so I'll make a note of that explanation there.

I've removed the incorrect paragraph as it's no longer relevant to the max message size.